src/libfreenect2.cpp

```diff
@@ -1115,4 +1119,239 @@ Freenect2Device *Freenect2::openDefaultDevice(const PacketPipeline *pipeline)
+}
+
+void Freenect2ReplayDevice::processDepthFrame(Frame* frame)
+{
+  DepthPacket packet;
+  packet.timestamp = frame->timestamp;
+  packet.sequence = frame->sequence;
+  packet.buffer = frame->data;
+  packet.buffer_length = frame->bytes_per_pixel;
+ 
+  pipeline_->getDepthPacketProcessor()->process(packet);
+}
+
+void Freenect2ReplayDevice::loadP0Tables(unsigned char* buffer, size_t buffer_length)
+{
+  pipeline_->getDepthPacketProcessor()->loadP0TablesFromCommandResponse(buffer, buffer_length);
+}
+
+void Freenect2ReplayDevice::loadXZTables(const float *xtable, const float *ztable)
+{
+  pipeline_->getDepthPacketProcessor()->loadXZTables(xtable, ztable);
+}
+
+void Freenect2ReplayDevice::loadLookupTable(const short *lut)
+{
+  pipeline_->getDepthPacketProcessor()->loadLookupTable(lut);
+}
+
+void runner (void* arg)
+{
+  static_cast<Freenect2ReplayDevice*>(arg)->run();
+}
+
+bool Freenect2ReplayDevice::start()
+{
+  running_ = true;
+  startStreams(true, true);
+  t_ = new thread(runner, this);
+  return running_;
+}
+
+bool Freenect2ReplayDevice::startStreams(bool enable_rgb, bool enable_depth)
+{
+  LOG_INFO << "starting...: rgb: " << enable_rgb << ", depth: " << enable_depth;
+
+  if (enable_rgb)
+  {
+    LOG_INFO << "starting rgb replay...";
+  }
+
+  if (enable_depth)
+  {
+    LOG_INFO << "submitting depth replay...";
+  }
+
+  LOG_INFO << "started";
+  return true;
+}
+
+bool Freenect2ReplayDevice::stop()
+{
+  running_ = false;
+  t_->join();
+  delete t_;
+  t_ = NULL;
+  return !running_;
+}
+
+bool hasSuffix(const std::string& str, const std::string& suffix)
+{
+  if (str.length() < suffix.length())
+  {
+    return false;
+  }
+  return str.compare(str.length() - suffix.length(), suffix.length(), suffix) == 0;
+}
+
+bool Freenect2ReplayDevice::run()
+{
+  DIR *d;
+  struct dirent *dir;
+
+  std::vector<std::string> frames;
+
+  d = opendir(directory_);
+  while ((dir = readdir(d)) != NULL)
+  {
+    std::string name = dir->d_name;
+    if (hasSuffix(name, ".depth")) {
+      frames.push_back(name);
+    }
+  }
+
+  for (size_t i = 0; i < frames.size() && running_; i++) {
+    std::string frame = frames[i];
+
+    size_t ix1 = frame.find("_");
+    size_t ix2 = frame.find("_", ix1 + 1);
+    size_t ix3 = frame.find(".", ix2 + 1);
+
+    std::string ts = frame.substr(0, ix1);
+    std::string seq = frame.substr(ix2 + 1, ix3);
+
+    std::string data;
+    if (hasSuffix(frame, ".depth")) {
+      std::ifstream fd(frame.c_str());
+      fd.seekg(0, fd.end);
+      int length = fd.tellg();
+      fd.seekg(0, fd.beg);
+
+      DepthPacket packet;
+      packet.timestamp = atoi(ts.c_str());
+      packet.sequence = atoi(seq.c_str());
+      packet.buffer = new unsigned char[length];
+      packet.buffer_length = length;
+
+      fd.read(reinterpret_cast<char*>(packet.buffer), length);
+      fd.close();
+
+      pipeline_->getDepthPacketProcessor()->process(packet);
+
+      delete[] packet.buffer;
+    }
+  }
+  
+  return true;
```

Comment:
No condition actually returns false. This function should return void.

Reply:
`void` it is.

DONE